career-link-webapi/Services/PostService.cs

```diff
@@ -70,8 +79,119 @@ public async Task<PostDTO> GetPostById(int id)
+                .Where(p => p.experienceLevel == experienceLevel)
+                .ToListAsync();
+
+            var filteredPosts = posts.Select(post => new
+            {
+                Post = post,
+                MatchingSkillsCount = post.Skills.Count(skill => skillIds.Contains(skill.SkillId))
+            }).Where(p => p.MatchingSkillsCount >= 2).OrderByDescending(p => p.MatchingSkillsCount).Select(p => p.Post).ToList();
+            var postDtos = filteredPosts.Select(post => new PostDTO
+            {
+                PostId = post.PostId,
+                Description = post.Description,
+                experienceLevel = post.experienceLevel,
+                CreatedDate = post.CreatedDate,
+                UpdatedDate = post.UpdatedDate,
+                User = new UserDTO
+                {
+                    UserId = post.User.Id,
+                    UserName = post.User.UserName ?? "Unknown",
+                    Email = post.User.Email ?? "Unknown",
+                    PhoneNumber = post.User.PhoneNumber ?? "Unknown",
+                    CreatedDate = post.User.CreatedDate,
+                    UpdatedDate = post.User.UpdatedDate
+                },
+                SkillNames = post.Skills.Select(s => s.Name).ToList()
+            }).ToList();
+
+            return postDtos;
+        }
+
+        public async Task<PostDTO> CreatePost(PostDTO postDto)
+        {
+            var user = await _dbContext.Users.FirstOrDefaultAsync(u => u.Id == postDto.User.UserId);
+            if (user == null)
+            {
+                throw new Exception("User not found");
+            }
+            var newPost = new Post
+            {
+                User = user,
+                Description = postDto.Description,
+                experienceLevel = postDto.experienceLevel,
+                CreatedDate = DateTime.Now,
+                UpdatedDate = DateTime.Now,
+            };
+            if (postDto.SkillIds != null && postDto.SkillIds.Count > 0)
+            {
+                newPost.Skills = await _dbContext.Skills.Where(skill => postDto.SkillIds.Contains(skill.SkillId)).ToListAsync();
+            }
+            _dbContext.Posts.Add(newPost);
+            await _dbContext.SaveChangesAsync();
+            return new PostDTO
+            {
+                PostId = newPost.PostId,
+                User = new UserDTO
+                {
+                    UserId = newPost.User.Id,
+                    UserName = newPost.User.UserName ?? "Unknown",
+                    Email = newPost.User.Email ?? "Unknown",
+                    PhoneNumber = newPost.User.PhoneNumber ?? "Unknown",
+                },
+                Description = newPost.Description,
+                experienceLevel = newPost.experienceLevel,
+                CreatedDate = newPost.CreatedDate,
+                UpdatedDate = newPost.UpdatedDate,
+                SkillIds = newPost.Skills.Select(s => s.SkillId).ToList()
+            };
```

Comment:
you dont need to return the whole dto, just return the name or simply a string message say success create is enough